includes/Gateway/Cash_App_Pay_Gateway.php

```diff
@@ -0,0 +1,1232 @@
+			}
+		}
+
+		return $order;
+	}
+
+	/**
+	 * Gets the configured environment ID.
+	 *
+	 * @since x.x.x
+	 *
+	 * @return string
+	 */
+	public function get_environment() {
+		return self::ENVIRONMENT_PRODUCTION;
+	}
+
+	/**
+	 * Gets the configured application ID.
+	 *
+	 * @since x.x.x
+	 *
+	 * @return string
+	 */
+	public function get_application_id() {
+		$square_application_id = 'sq0idp-wGVapF8sNt9PLrdj5znuKA';
+
+		if ( $this->get_plugin()->get_settings_handler()->is_sandbox() ) {
+			$square_application_id = $this->get_plugin()->get_settings_handler()->get_option( 'sandbox_application_id' );
+		}
+
+		/**
+		 * Filters the configured application ID.
+		 *
+		 * @since x.x.x
+		 *
+		 * @param string $application_id application ID
+		 */
+		return apply_filters( 'wc_square_application_id', $square_application_id );
+	}
+
+	/**
+	 * Returns the $order object with a unique transaction ref member added
+	 *
+	 * @since x.x.x
+	 * @param WC_Order_Square $order the order object
+	 * @return WC_Order_Square order object with member named unique_transaction_ref
+	 */
+	protected function get_order_with_unique_transaction_ref( $order ) {
+		$order_id = $order->get_id();
+
+		// generate a unique retry count
+		if ( is_numeric( $this->get_order_meta( $order_id, 'retry_count' ) ) ) {
+			$retry_count = $this->get_order_meta( $order_id, 'retry_count' );
+			++$retry_count;
+		} else {
+			$retry_count = 0;
+		}
+
+		// keep track of the retry count
+		$this->update_order_meta( $order, 'retry_count', $retry_count );
+
+		$order->unique_transaction_ref = time() . '-' . $order_id . ( $retry_count >= 0 ? '-' . $retry_count : '' );
+		return $order;
+	}
+
+	/**
+	 * Returns the payment method image URL (if any) for the given $type, ie
+	 * if $type is 'amex' a URL to the american express card icon will be
+	 * returned.  If $type is 'echeck', a URL to the echeck icon will be
+	 * returned.
+	 *
+	 * @since x.x.x
+	 * @param string $type the payment method cc type or name
+	 * @return string the image URL or null
+	 */
+	public function get_payment_method_image_url( $type = '' ) {
+		/**
+		 * Payment Gateway Fallback to PNG Filter.
+		 *
+		 * Allow actors to enable the use of PNGs over SVGs for payment icon images.
+		 *
+		 * @since x.x.x
+		 * @param bool $use_svg true by default, false to use PNGs
+		 */
+		$image_extension = apply_filters( 'wc_payment_gateway_' . $this->get_id() . '_use_svg', true ) ? '.svg' : '.png';
+
+		// first, is the card image available within the plugin?
+		if ( is_readable( $this->get_plugin()->get_plugin_path() . '/assets/images/cash-app' . $image_extension ) ) {
+			return \WC_HTTPS::force_https_url( $this->get_plugin()->get_plugin_url() . '/assets/images/cash-app' . $image_extension );
+		}
+
+		// Fall back to framework image URL.
+		return parent::get_payment_method_image_url( $type );
+	}
```

Comment:
I suspect there isn't a lot that can be done about it here but this function is a little confusing in the Cash App context because there is no need for the `type` parameter but it is required because of the inheritance structure and how it's defined in the parent class. 


Not a blocker at all but perhaps that can be addressed by updating the function block comment so that is clear. The current one looks like a copy of the base class and refers to Amex etc. 

Ie something along the lines of:


```php
/**
 * Returns the payment method image URL (if any).
 *
 * The type parameter is unused in this gateway's context because there aren't other types that it supports. 
 * 
 * @since x.x.x
 * @param string $type the payment method cc type or name
 * @return string the image URL or null
 */
public function get_payment_method_image_url( $type = '' ) {
```

Reply:
The inline comment (see below) in this function should also be updated as it references "card" which doesn't apply here.

```
// first, is the card image available within the plugin?
```

Reply:
Thanks for noting this. handled in https://github.com/woocommerce/woocommerce-square/pull/53/commits/9d3bc1f25a86ccb73e98216aabc09fb13fa9baec